src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTabMenu.java

```diff
@@ -70,21 +69,40 @@
 
     public GuiTabMenu(GuiContainer guiContainer)
     {
-        this.guiContainer = guiContainer;
-        this.guiContainerClass = guiContainer.getClass();
-        // get a copy of the stored list so it can be modified without consequence
-        tabs = new ArrayList<>();
-        List<GuiTab> tabsForGui = GuiTab.getTabsForGui(this.guiContainer.getClass());
-        if (guiContainer instanceof GuiInventory)
+        tabs = new ArrayList<>(); // get a copy of the stored list so it can be modified without consequence
+        if (guiContainer instanceof IGuiTabAttachable)
         {
+            IGuiTabAttachable attachable = ((IGuiTabAttachable) guiContainer);
+            // get a copy of the stored list so it can be modified without consequence
+            this.guiContainerClass = attachable.getMainGuiContainer().getClass();
+            List<GuiTab> tabsForGui = GuiTab.getTabsForGui(this.guiContainer.getClass());
+            GuiTab defaultTab;
```

Comment:
should the default gui be converted into a private field and have a private setter and a public getter ?